gems/pending/appliance_console/logfile_configuration.rb

```diff
@@ -1,45 +1,103 @@
 require 'linux_admin'
 require 'pathname'
 require 'fileutils'
+require 'util/miq-system.rb'
 require 'appliance_console/logical_volume_management'
 require 'appliance_console/prompts'
 
 module ApplianceConsole
   class LogfileConfiguration
     LOGFILE_DIRECTORY = Pathname.new("/var/www/miq/vmdb/log").freeze
     LOGFILE_NAME = "miq_logs".freeze
+    MIQ_LOGS_CONF = Pathname.new("/etc/logrotate.d/miq_logs.conf").freeze
 
-    attr_accessor :disk
+    attr_accessor :size, :disk, :logrotate_count,
+                  :requested_new_disk, :activated_new_disk,
+                  :requested_new_logrotate_count, :activated_new_logrotate_count
 
     include ApplianceConsole::Logging
 
+    def initialize
+      self.requested_new_disk            = false
+      self.activated_new_disk            = false
+      self.requested_new_logrotate_count = false
+      self.activated_new_logrotate_count = false
+
+      self.size = MiqSystem.disk_usage(LOGFILE_DIRECTORY)[0][:total_bytes]
+      self.logrotate_count = File.readlines(MIQ_LOGS_CONF).detect { |line| line =~ /rotate\s*\d.*$/ }.split(" ")[1].to_i
+    end
+
     def activate
-      stop_evm
-      initialize_logfile_disk
-      start_evm
+      activated_new_disk = activate_new_disk
+      activated_new_logrotate_count = activate_new_logrotate_count
+
+      activated_new_disk || activated_new_logrotate_count
     end
 
     def ask_questions
       clear_screen
-      return false unless use_new_disk
-      choose_disk
+      choose_disk if use_new_disk
+      choose_logrotate_count if set_new_logrotate_count
       confirm_selection
     end
 
     private
 
     def confirm_selection
-      agree("Continue with disk: #{disk.path}: #{disk.size.to_i / 1.megabyte} MB (Y/N):")
+      return false unless requested_new_disk || requested_new_logrotate_count
+
+      clear_screen
+      if requested_new_disk
+        say("\t#{disk.path} with #{disk.size.to_i / 1.megabyte} MB will be configured as the new logfile disk.")
+      end
+
+      if requested_new_logrotate_count
+        say("\tThe number of saved logratations will be updated to: #{logrotate_count}")
+      end
+
+      agree("Confirm continue with these updates (Y/N):")
     end
 
     def use_new_disk
-      agree("Configure a new logfile disk volume? (Y/N):")
+      self.requested_new_disk = agree("Configure a new logfile disk volume? (Y/N):")
     end
 
     def choose_disk
       self.disk = ask_for_disk("logfile disk")
     end
 
+    def set_new_logrotate_count
+      self.requested_new_logrotate_count = agree("Change the saved logrotate count from #{logrotate_count}? (Y/N):")
+    end
+
+    def choose_logrotate_count
+      say "\t1000 MBs of disk space is recommended for each saved log rotation."
```

Comment:
@carbonin Because throughout the appliance_console disk size is reported in MB. So I was just trying to be consistent.